test/compiler.test.js

```diff
@@ -1096,6 +1193,38 @@ describe('compiler', function() {
         'sourceFile', moduleJS);
     });
 
+    it('loads middleware from coffeescript', function() {
+      require('coffee-script/register');
+      var coffee = appdir.writeFileSync('my-middleware.coffee', '');
+      appdir.writeConfigFileSync('middleware.json', {
+        'routes': {
+          './my-middleware': {}
+        }
+      });
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.middleware.middleware[0]).have.property(
+        'sourceFile', coffee);
+    });
+
+    it('prefers coffeescript over json when loading middleware', function() {
+      require('coffee-script/register');
+      var coffee = appdir.writeFileSync('my-middleware.coffee', '');
+      appdir.writeFileSync('my-middleware.json', '');
+      appdir.writeConfigFileSync('middleware.json', {
+        'routes': {
+          './my-middleware': {}
+        }
+      });
+
+      var instructions = boot.compile(appdir.PATH);
+
+      expect(instructions.middleware.middleware).to.have.length(1);
+      expect(instructions.middleware.middleware[0]).have.property(
+        'sourceFile', coffee);
+    });
```

Comment:
Please add another unit-test to verify that coffeescript files are loaded from `node_modules` too, i.e. create `node_modules/my-middleware/index.coffee` (+json) instead of `my-middleware.coffee` (+ json).